server/plugin/client.go

```diff
@@ -179,6 +179,12 @@ func (c *client) call(basePath, method, path, contentType string, mattermostUser
 		req.Header.Add("Content-Type", contentType)
 	}
 
+	if isAccessTokenExpired := c.plugin.isAccessTokenExpired(mattermostUserID); isAccessTokenExpired {
```

Comment:
You can make this function return the refresh token also, so that you won't have to load the user again from the DB inside the `RefreshOAuthToken` function.